File: Lab Report 2/C++/Task2.cpp
#include<iostream>
using namespace std;

class complex {
    private:
        double re, im;
    public:
        complex();
        complex(double r, double i);
        // complex negate();       //returns complex object
        void show();
        void input();
        complex addCom(complex c);
        complex subCom(complex c);
        complex mulCom(complex c);
};

complex::complex():re(0),im(0){}

complex::complex(double r, double i):re(r),im(i){}

void complex::show() {
    if(im>0)
        cout << re << "+" << im << "i" << endl;
    else
        cout << re << im << "i" << endl;
}

void complex::input() {
    cout<<"Enter real number: ";
    cin>>re;
    cout<<"Enter imaginary number: ";
    cin>>im;
}

complex complex::addCom(complex c) {
    complex x;
    x.re = re + c.re;
    x.im = im + c.im;
    return x;    
}

complex complex::subCom(complex c) {
    complex x;
    x.re = re - c.re;
    x.im = im - c.im;
    return x;    
}

complex complex::mulCom(complex c) {
    complex x;
    x.re = (re * c.re) - (im * c.im);
    x.im = (re * c.im) + (c.re * im);
}

main() {
    complex c1,c2,c;
    
    c1.input();
    cout<<"c1 = ";
    c1.show();

    c2.input();
    cout<<"c2 = ";
    c2.show();

    c = c1.addCom(c2);
    cout<<"Sum = ";
    c.show();

    c = c1.subCom(c2);
    cout<<"Difference = ";
    c.show();

    c = c1.mulCom(c2);
    cout<<"Product = ";
    c.show();
}

File: Lab Report 2/Python/Task2.py
class Complex:
    re = 0
    im = 0

    def __init__(self):
        self.re = 0
        self.im = 0

    def __init__(self, r, i):
        self.re = r
        self.im = i

    def show(self):
        if self.im >= 0:
            print(f"{self.re}+{self.im}i")
        else:
            print(self.re, self.im, "i", sep="")

    def addCom(self, c1):
        c = Complex(0,0)
        c.re = self.re + c1.re
        c.im = self.im + c1.im
        return c

    def subCom(self, c1):
        c = Complex(0,0)
        c.re = self.re - c1.re
        c.im = self.im - c1.im
        return c

    def mulCom(self, c1):
        c = Complex(0,0)
        c.re = (c1.re * self.re) - (c1.im * self.im)
        c.im = (c1.re * self.im) + (self.re * c1.im)
        return c

    def input(self):
        self.re = int(input("Enter real number: "))
        self.im = int(input("Enter imaginary number: "))


c1 = Complex(0,0)
c2 = Complex(0,0)
c = Complex(0,0)

print("For Complex Number C1:")
c1.input()
print("C1:", end = " ")
c1.show()

print("For Complex Number C2:")
c2.input()
print("C2:", end = " ")
c2.show()

print("Sum =", end = " ")
c = c1.addCom(c2)
c.show()

print("Difference =", end = " ")
c = c1.subCom(c2)
c.show()

print("Product =", end = " ")
c = c1.mulCom(c2)
c.show()